src/main/resources/assets/tconstruct/book/puny_smelting/sections/upgrades.json

```diff
@@ -144,6 +148,51 @@
     "data": "upgrades/harvest/lightspeed.json"
   },
   
+  {
+    "name": "group_ranged",
+    "type": "mantle:text",
+    "data": "upgrades/ranged.json"
+  },
```

Comment:
In the encyclopedia, I relisted some modifiers from melee which have different effects on bows, notably piercing, necrotic, and fiery. I would recommend doing the same here as its not obvious how you get things like flame otherwise.